src/typescriptGenerator.ts

```diff
@@ -78,7 +78,7 @@ export async function generateTypeScriptLibrary(
   // Skip metadata generation if `generate-metadata` is explicitly false
   generatePackageJson(project, clientDetails);
   generateLicenseFile(project);
-  generateReadmeFile(clientDetails, project);
+  generateReadmeFile(codeModel, project);

```

Comment:
readmeFileGenerator is shared by RLC and HLC, there is no ClientDetails in RLC, so I changed the parameter from ClientDetails to CodeModel

Reply:
If possible we should avoid passing the whole codeModel to the generators. Can we just pass the specific data to generateReadmeFile instead of the whole codeModel?

Reply:
sure